charm/kafka/reactive/storage.py

```diff
@@ -9,6 +9,18 @@
 from charmhelpers.core.hookenv import log
 
 
+def init_brokerid(log_dir):
+    broker_path = os.path.join(log_dir, '.broker_id')
+    storageids = hookenv.storage_list('logs')
+    broker_id = storageids[0].split('/')[1]
```

Comment:
This could cause a broker_id collision, if some brokers have storage attached and others don't, at any point in time, because you're using two ID schemes which could overlap:
- Numeric suffix of Unit ID
- Numeric suffix of storage ID

There's no guarantee that these numbers will be unique. Imagine a case where a kafka broker is deployed, related to the cluster, and then storage is attached. When deployed from a bundle, you can't depend on the order in which these operations will be applied.